lib/identity_cache/query_api.rb

```diff
@@ -182,7 +182,12 @@ def add_cached_associations_to_coder(record, coder)
           end
           if (cached_has_manys = klass.cached_has_manys).present?
             coder[:association_ids] = cached_has_manys.each_with_object({}) do |(name, options), hash|
-              hash[name] = record.instance_variable_get(:"@#{options[:ids_variable_name]}") unless options[:embed] == true
+              unless options[:embed] == true
+                iv_name = :"@#{options[:ids_variable_name]}"
+                hash[name] = if record.instance_variable_defined?(iv_name)
```

Comment:
It looks like this check is unnecessary, and was just added to workaround a warning in another specific test. SchemaChangeTest#test_add_non_embedded_cache_has_many does `Item.expects(:resolve_cache_miss).returns(@record)` which mocks out the code that sets this variable.